connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,112 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.*;
+import java.lang.reflect.InvocationTargetException;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ConnectionTest {
+    private static Connection connection;
+    private static ByteArrayOutputStream innerOutputStream;
+    private static int originalNumberOfBytes;
+    private static final SocketAddress address = mock(SocketAddress.class);
+    private static final Message messageToReceive = new Message(TEXT, "MessageToReceive");
+    private static final Message messageToSend = new Message(TEXT, "MessageToSend");
+    @Rule
+    public final ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void setUpConnectionObjectWithByteStreams() throws IOException {
+        ByteArrayOutputStream outerOutputStream = new ByteArrayOutputStream();
+        try (ObjectOutputStream outerOOS = new ObjectOutputStream(outerOutputStream)) {
+            outerOOS.writeObject(messageToReceive);
+        }
+        ByteArrayInputStream innerInputStream = new ByteArrayInputStream(outerOutputStream.toByteArray());
```

Comment:
&#x1F34E; **Issue**
You should initialize variable at the closest point possible to its intended usage.